src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1487 @@
+            # Note: floating point overflow is an issue if variance weights
+            # are used - it can't be squared before multiplication
+            if rn_var is not None:
+                # Add the inverse of the resampled variance to a running sum.
+                # Update only pixels (in the running sum) with
+                # valid new values:
+                mask = (rn_var >= 0) & np.isfinite(rn_var) & (weight > 0)
+                self._var_rnoise_wsum[mask] = np.nansum(
+                    [
+                        self._var_rnoise_wsum[mask],
+                        rn_var[mask] * weight[mask] * weight[mask]
+                    ],
+                    axis=0
+                )
+                self._var_rnoise_weight[mask] += weight[mask]
+
+        # Now do poisson and flat variance, updating only valid new values
+        # (zero is a valid value; negative, inf, or NaN are not)
+        if self._check_var_array(model, "var_poisson"):
+            pn_var = self._resample_one_variance_array(
+                "var_poisson",
+                model=model,
+                **pars,
+            )
+            if pn_var is not None:
+                mask = (pn_var >= 0) & np.isfinite(pn_var) & (weight > 0)
+                self._var_poisson_wsum[mask] = np.nansum(
+                    [
+                        self._var_poisson_wsum[mask],
+                        pn_var[mask] * weight[mask] * weight[mask]
+                    ],
+                    axis=0
+                )
+                self._var_poisson_weight[mask] += weight[mask]
+
+        if self._check_var_array(model, "var_flat"):
+            flat_var = self._resample_one_variance_array(
+                "var_flat",
+                model=model,
+                **pars,
+            )
+            if flat_var is not None:
+                mask = (flat_var >= 0) & np.isfinite(flat_var) & (weight > 0)
+                self._var_flat_wsum[mask] = np.nansum(
+                    [
+                        self._var_flat_wsum[mask],
+                        flat_var[mask] * weight[mask] * weight[mask]
+                    ],
+                    axis=0
+                )
+                self._var_flat_weight[mask] += weight[mask]
+
+    def finalize_resample_variance(self, output_model, free_memory=True):
+        """ Compute variance for the resampled image from running sums and
+        weights. Free memory (when ``free_memory=True``) that holds these
+        running sums and weights arrays.
+        """
+        # Divide by the total weights, squared, and set in the output model.
+        # Zero weight and missing values are NaN in the output.
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", "invalid value*", RuntimeWarning)
+            warnings.filterwarnings("ignore", "divide by zero*", RuntimeWarning)
+
+            output_variance = (
+                self._var_rnoise_wsum / self._var_rnoise_weight /
```

Comment:
A double division `x/y/z` is equivalent to `x / (y * z)`.  Since divisions are computationally expensive compared to multiplications, I recommend using the latter format.  And, with the parentheses, it's easier to read.